Branch-SDK/src/io/branch/referral/util/BranchViewHandler.java

```diff
@@ -0,0 +1,334 @@
+            showBranchView(branchView, currentActivity, callback);
+        }
+
+        return isBranchViewDialogShowing_;
+    }
+
+    private void showBranchView(final BranchView branchView, Activity currentActivity, final IBranchViewEvents callback) {
+        if (currentActivity != null && branchView != null) {
+            branchView.updateUsageCount(currentActivity.getApplicationContext(), branchView.branchViewID_);
+            WebView webView = new WebView(currentActivity);
+
+            final RelativeLayout layout = new RelativeLayout(currentActivity);
+            layout.setVisibility(View.GONE);
+
+            RelativeLayout.LayoutParams layoutParams = new RelativeLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT);
+            layoutParams.setMargins(40, 80, 40, 80);
+
+            layout.addView(webView, layoutParams);
+            layout.setBackgroundColor(Color.parseColor("#11FEFEFE"));
+
+
+            if (Build.VERSION.SDK_INT >= 19) {
+                webView.setLayerType(View.LAYER_TYPE_HARDWARE, null);
+            }
+
+            if (!TextUtils.isEmpty(branchView.webViewUrl_)) {
+                webView.loadUrl(branchView.webViewUrl_);
+            } else if (!TextUtils.isEmpty(branchView.webViewHtml_)) {
+                webView.loadDataWithBaseURL(null, branchView.webViewHtml_, "text/html", "utf-8", null);
+            } else {
+                return; // Error no url or Html
+            }
+            isBranchViewDialogShowing_ = true;
+            final Dialog dialog = new Dialog(currentActivity, android.R.style.Theme_Black_NoTitleBar_Fullscreen);
+            dialog.setContentView(layout);
+            dialog.show();
+
+            if (callback != null) {
+                callback.onBranchViewVisible(branchView.branchViewAction_);
+            }
+
+            webView.getSettings().setJavaScriptEnabled(true);
+
+            webView.setWebViewClient(new WebViewClient() {
+                @Override
+                public boolean shouldOverrideUrlLoading(WebView view, String url) {
+                    boolean isHandled = handleUserActionRedirect(url);
+                    if (!isHandled) {
+                        view.loadUrl(url);
+                    } else {
+                        dialog.dismiss();
+                    }
+                    return isHandled;
+                }
+
+                @Override
+                public void onPageFinished(WebView view, String url) {
+                    super.onPageFinished(view, url);
+                    layout.setVisibility(View.VISIBLE);
+                    view.setVisibility(View.VISIBLE);
+                    dialog.show();
+                    showViewWithAlphaAnimation(layout);
+                    showViewWithAlphaAnimation(view);
+                }
+            });
+
+            dialog.setOnDismissListener(new DialogInterface.OnDismissListener() {
```

Comment:
since a dialog can handle onBackPressed - do we wanna hook into that and update callbacks accordingly? or is onDismiss sufficient?